packages/lib-classifier/src/store/Classification.js

```diff
@@ -46,6 +47,38 @@ const Classification = types
     }),
     metadata: types.maybe(ClassificationMetadata)
   })
+  .views(self => ({
+    get toSnapshot () {
+      let snapshot = getSnapshot(self)
+      let annotations = []
+      self.annotations.forEach(annotation => {
+        annotations = annotations.concat(annotation.toSnapshot)
+      })
+      snapshot = Object.assign({}, snapshot, { annotations })
+      return snapshot
+    }
+  }))
+  .preProcessSnapshot(snapshot => {
+    const newSnapshot = Object.assign({}, snapshot)
+    // generate classification IDs, if not present
+    newSnapshot.id = snapshot.id || cuid()
+    // convert any annotations arrays to a map
+    const annotationsMap = {}
+    if (snapshot.annotations && Array.isArray(snapshot.annotations)) {
+      snapshot.annotations.forEach(annotation => annotationsMap[annotation.task] = annotation)
```

Comment:
The naming of `annotationsMap` is a little misleading because it's not actually being converted into a map here, just a object. Since order of the annotations is important, how about converting into an array of pairs:

`const arrayOfPairs = [['T1', { taskType: 'single', value: 0 }], ['T2', { taskType: 'text', value: 'hello' }]]` 

then use `new Map(arrayOfPairs)`?

Reply:
Worth trying but this is creating the new snapshot that we then pass into `classification.create(snapshot)` during `createClassification()`.

Reply:
I think MST accepts an array of pairs as a snapshot for a new map, but I'll double check.

I could add tests for this as well. It was low priority for me because we aren't actually loading in an incomplete classification from Panoptes yet.

Reply:
Transforming the input snapshot to an array of pairs gives this error 😞 
```
at path "/annotations" snapshot 
`[["T1",{"task":"T1","taskType":"single","value":1}],
["T0",{"task":"T0","taskType":"text","value":"Hello"}]]` 
is not assignable to type: 
`map<string, (DataVisAnnotation | DrawingAnnotation | 
MultipleChoiceAnnotation | SingleChoiceAnnotation | 
TextAnnotation)>` (Value is not a plain object)
```

I could have sworn that the MST docs say you can create a map from an array of `[key, value]` pairs.

Reply:
Using a map
```
Map {
  'T1' => { task: 'T1', taskType: 'single', value: 1 },
  'T0' => { task: 'T0', taskType: 'text', value: 'Hello' }
}
```
also throws during `classification.create()`.

```
at path "/annotations" snapshot `{}` is not assignable to type: 
`map<string, (DataVisAnnotation | DrawingAnnotation | MultipleChoiceAnnotation | 
SingleChoiceAnnotation | TextAnnotation)>` (Value is not a plain object),
```

Reply:
Oooh, annotations are mapped by `annotation.id` now, not `annotation.task`. This is going to be interesting, since we strip IDs when we save to Panoptes.

Reply:
This hasn't been a problem for us to date because we only ever create classifications with an empty annotations collection.

Reply:
I've added a couple of mock Panoptes annotations to the classification tests, along with a test to check that the array order is preserved after `classification.create(snapshot)`.